misk/src/main/kotlin/misk/hibernate/Transacter.kt

```diff
@@ -1,5 +1,23 @@
 package misk.hibernate
 
+/**
+ * Provides explicit block-based transaction demarcation.
+ */
 interface Transacter {
+  /**
+   * @returns [true] if the calling thread is currently within a transaction block.
+   */
+  val inTransaction: Boolean
+
+  /**
+   * Executes [lambda] in a transaction. The transaction will be committed once [lambda] completes.
+   * If [lambda] raises an exception the transaction will be rolled back.
+   *
+   * @param lambda a function to execute in a transaction.
```

Comment:
nit: this doc doesn’t add anything.
https://publicobject.com/2007/08/05/stop_writing_redundant_javadoc/